internal/collections/map_test.go

```diff
@@ -106,3 +106,26 @@ func TestNewCaseSensitiveKeyMap(t *testing.T) {
 	}
 
 }
+
+func BenchmarkTxSetGet(b *testing.B) {
+	// Pre-generate keys in a map to avoid fmt.Sprintf overhead during benchmarking
+	keys := make(map[int]string, b.N)
+	for i := 0; i < b.N; i++ {
+		keys[i] = fmt.Sprintf("key%d", i)
+	}
+	c := NewCaseSensitiveKeyMap(variables.RequestHeaders)
+	b.Run("Set", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			c.Set(keys[i], []string{"value2"})
+		}
+	})
+	b.Run("Get", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			c.Set(keys[i], []string{"value2"})
```

Comment:
since you already set the values before, shall we do you do this here again? also, likely we reset the benchmarks after this to not to pollute what we are measuring.

Reply:
Done, fixed it.